packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -73,4 +85,31 @@ contract L1Block is ISemver {
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
     }
+
+    /// @notice Updates the L1 block values for a post-blob activated chain.
+    /// Params are passed in as part of msg.data in order to compress the calldata.
+    /// Params should be passed in in the following order:
+    ///   1. _basefeeScalar      L1 basefee scalar
+    ///   2. _blobBasefeeScalar  L1 blob basefee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _number             L1 blocknumber.
+    ///   5. _timestamp          L1 timestamp.
+    ///   6. _basefee            L1 basefee.
+    ///   7. _blobBasefee        L1 blobBasefee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    function setL1BlockValuesEcotone() external {
+        require(msg.sender == DEPOSITOR_ACCOUNT, "L1Block: only the depositor account can set L1 block values"); 
+        uint256 data;
+        assembly {
+            data := calldataload(4)
+            sstore(basefeeScalar.slot, shr(192, data)) // basefeeScalar & blobBaseFeeScalar (uint32,uint32)
+            sstore(sequenceNumber.slot, shr(128, data)) // uint64
+            sstore(number.slot, shr(128, calldataload(20))) // number & timestamp (uint64,uint64)
+            sstore(basefee.slot, calldataload(36))  // uint256
+            sstore(blobBasefee.slot, calldataload(68)) // uint256
+            sstore(hash.slot, calldataload(100)) // bytes32
+            sstore(batcherHash.slot, calldataload(132)) // bytes32
+	    }
```

Comment:
looks like this needs a linting

Reply:
yeah poor editor config... should be fixed.